nixos/doc/manual/development/option-types.section.md

```diff
@@ -200,9 +200,11 @@ Submodules are detailed in [Submodule](#section-option-types-submodule).
 `types.submodule` *`o`*
 
 :   A set of sub options *`o`*. *`o`* can be an attribute set, a function
-    returning an attribute set, or a path to a file containing such a
-    value. Submodules are used in composed types to create modular
-    options. This is equivalent to
+    returning an attribute set, a list of attribute sets, or a path to a
+    file containing such a value. When the argument is a list, it is
+    equivalent to the argument being a set with the argument as the value
+    of the `imports` attribute. Submodules are used in composed types to
+    create modular options. This is equivalent to
```

Comment:
I think this entire paragraph warrants a rewrite, here's my suggestion:

> `types.submodule` *`module`*: A nested module evaluation, containing its own set of options. `module` can either be a single module (in the form of an attribute set, a function, or a path), or a list of modules (which then gets processed just like a list of `imports`). Submodules can be used in composed types to create modular options. This type is equivalent to

Reply:
I agree that reads better. For me as a newcomer it seems hard to tell whether "a list of `imports`" refers to `imports` or `import`. Maybe this could be clarified using
> which then gets processed just like a list assigned to `imports`